HCSP.js

```diff
@@ -0,0 +1,228 @@
+    // TODO: Find a way to have a single field instead of two
+    var creator = ZU.cleanAuthor('HCSP', 'author', false);
+    newItem.creators.push(creator);
+
+    /*---------- URL ----------*/
+    newItem.url = doc.location.href;
+    
+    /*---------- Title ----------*/
+    var xpathTitle = '//li[@class="active"]';
+    var title = ZU.xpathText(doc, xpathTitle);
+    newItem.title = title;
+    Zotero.debug(title);
+ 
+    /*---------- Date ----------*/
+    // Date is not easy to find because it is not in a named bloc. It is
+    // somewhere in one of the <p> in the div 'doc-donnees'
+    // Then to find it, use a Regex on each <p> bloc
+    var xpathDate = "//div[@id='doc-donnees']/p";
+    var docDonnees = ZU.xpath(doc, xpathDate);
+
+    // test each <p> bloc (normally it's in the first one)
+    for (var nextP of docDonnees) {
+        var pContent = nextP.textContent;
+
+        if (pContent.match("Date du document")){
+            var date = pContent.match("[0-9]{2}/[0-9]{2}/[0-9]{4}");
+            break;
+        }
+    }
+    
+    newItem.date = String(date);
+
+    /*---------- Tags ----------*/
+    var xpathTags = '//div[@id="doc-donnees"]/span/a';
+    var xresultTags = ZU.xpath(doc, xpathTags);
+
+    for (var nextResult of xresultTags) {
+        var actualTag = nextResult.textContent;
+        var actualTagTrimmed = actualTag.trim();
+        newItem.tags.push(actualTagTrimmed);
+    }
+
+    /* --------- abstract -----------*/
+    var xpath_abstract = '//div[@class="avistexte"]';
+    var abstract_note = ZU.trimInternal(ZU.xpathText(doc, xpath_abstract, namespace, "\n")); 
+    newItem.abstractNote = abstract_note;
+
+    /* ----------- attachment --------- */
+    // PDFs are in a div on the right part of the page.
+    var xpath_pdf = '//div[@id="doc-donnees"]/p/a';
+    // ZU.xpath produce Arrays of HTMLAnchorElement
+    var xresult_pdfs = ZU.xpath(doc, xpath_pdf);
+
+    // The name of the PDF start after an "="
+    // Because there is no positive lookbehind in javascript, one should choose 
+    // the second element (index = 1)
+    var pdf_filename_regex = /=(.*)/
+
+    // For each PDF anchor get the url and title and push it
+    for (one_pdf_anchor of xresult_pdfs) {
+        var pdf_url = one_pdf_anchor.href;
+        var pdf_title = pdf_url.match(pdf_filename_regex)[1];
+
+        var one_pdf_attachement = {
+            url: pdf_url,
+            title: pdf_title,
+            mimeType: "application/pdf"
+        }
+
+        newItem.attachments.push(one_pdf_attachement);
+    }
+
+    newItem.complete();
+}
+
+/** BEGIN TEST CASES **/
+var testCases = [
+    {
+        "type": "web",
+        "url": "http://www.hcsp.fr/explore.cgi/avisrapportsdomaine?clefr=509",
+        "items": [
+            {
+                "itemType": "report",
+                "title": "Évaluation des besoins des personnes handicapées et  élaboration des réponses au sein des MDPH",
```

Comment:
looks like this needs cleaning up using ```ZU.trimInternal()``` -- there's a double space between "et" and "elaboration"